From 67a064ae0d959541375eef2181d171e97cb18d1b Mon Sep 17 00:00:00 2001
From: linuxx <io.nolinuxnoparty@gmail.com>
Date: Sun, 4 Jan 2015 22:49:41 +0100
Subject: [PATCH 6/7] Term: materialize

* Material UI
* Vectors

Change-Id: Ia5c3c9027662edd51021bf2183cd203d25bb7474

Terminal: all menu items should have their proper icon

Change-Id: I1ebb1b3cc612bfb3b2e6c72c2528ac76aee96212
Signed-off-by: Joey Rizzoli <joey@lineageos.org>

Terminal: update main theme color for improved readability

Change-Id: Ib8866c9f9c72e7f8104c069f49610fb9ffbec222
Signed-off-by: Joey Rizzoli <joey@lineageos.org>
---
 AndroidManifest.xml                           |  4 ++-
 res/drawable/ic_menu_add.xml                  | 13 ++++++++
 res/drawable/ic_menu_close_clear_cancel.xml   | 13 ++++++++
 res/drawable/ic_menu_settings.xml             | 25 ++++++++++++++
 res/layout/activity.xml                       | 33 ++++++++++++++-----
 res/menu/activity.xml                         | 10 +++---
 res/values/arrays.xml                         | 12 ++++---
 res/values/colors.xml                         | 23 +++++++++++++
 res/values/styles.xml                         | 26 +++++++++++++++
 res/xml/settings.xml                          |  3 +-
 .../android/terminal/TerminalActivity.java    |  8 +++++
 .../terminal/TerminalSettingsActivity.java    |  8 ++++-
 src/com/android/terminal/TerminalView.java    |  6 ++--
 13 files changed, 161 insertions(+), 23 deletions(-)
 create mode 100644 res/drawable/ic_menu_add.xml
 create mode 100644 res/drawable/ic_menu_close_clear_cancel.xml
 create mode 100644 res/drawable/ic_menu_settings.xml
 create mode 100644 res/values/colors.xml
 create mode 100644 res/values/styles.xml

diff --git a/AndroidManifest.xml b/AndroidManifest.xml
index e5f1956..a9fb4e3 100644
--- a/AndroidManifest.xml
+++ b/AndroidManifest.xml
@@ -37,7 +37,8 @@
 
         <activity
             android:name=".TerminalActivity"
-            android:windowSoftInputMode="stateAlwaysVisible|adjustResize">
+            android:windowSoftInputMode="stateAlwaysVisible|adjustResize"
+            android:theme="@style/TermTheme.NoActionBar">
             <intent-filter>
                 <action android:name="android.intent.action.MAIN" />
                 <category android:name="android.intent.category.DEFAULT" />
@@ -49,6 +50,7 @@
         <activity
             android:name=".TerminalSettingsActivity"
             android:label="@string/menu_item_settings"
+            android:theme="@style/TermTheme"
             android:excludeFromRecents="true">
             <intent-filter>
                 <action android:name="android.intent.action.MAIN" />
diff --git a/res/drawable/ic_menu_add.xml b/res/drawable/ic_menu_add.xml
new file mode 100644
index 0000000..b8f2af8
--- /dev/null
+++ b/res/drawable/ic_menu_add.xml
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="24dp"
+    android:height="24dp"
+    android:viewportWidth="24"
+    android:viewportHeight="24">
+
+    <path
+        android:fillColor="@color/white"
+        android:pathData="M19,13 L13,13 L13,19 L11,19 L11,13 L5,13 L5,11 L11,11 L11,5 L13,5 L13,11 L19,11 L19,13 Z" />
+    <path
+        android:pathData="M0,0 L24,0 L24,24 L0,24 Z" />
+</vector>
diff --git a/res/drawable/ic_menu_close_clear_cancel.xml b/res/drawable/ic_menu_close_clear_cancel.xml
new file mode 100644
index 0000000..469461d
--- /dev/null
+++ b/res/drawable/ic_menu_close_clear_cancel.xml
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="24dp"
+    android:height="24dp"
+    android:viewportWidth="24"
+    android:viewportHeight="24">
+
+    <path
+        android:fillColor="@color/white"
+        android:pathData="M19,6.41 L17.59,5 L12,10.59 L6.41,5 L5,6.41 L10.59,12 L5,17.59 L6.41,19 L12,13.41 L17.59,19 L19,17.59 L13.41,12 Z" />
+    <path
+        android:pathData="M0,0 L24,0 L24,24 L0,24 Z" />
+</vector>
diff --git a/res/drawable/ic_menu_settings.xml b/res/drawable/ic_menu_settings.xml
new file mode 100644
index 0000000..46a0136
--- /dev/null
+++ b/res/drawable/ic_menu_settings.xml
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="24dp"
+    android:height="24dp"
+    android:viewportWidth="24"
+    android:viewportHeight="24">
+
+    <path
+        android:fillColor="@color/white"
+        android:pathData="M19.43,12.98 C19.47,12.66,19.5,12.34,19.5,12 S19.47,11.34,19.43,11.02
+L21.54,9.37 C21.73,9.22,21.78,8.95,21.66,8.73 L19.66,5.27
+C19.54,5.05,19.27,4.97,19.05,5.05 L16.56,6.05 C16.04,5.65,15.48,5.32,14.87,5.07
+L14.49,2.42 C14.46,2.18,14.25,2,14,2 L10,2 C9.75,2,9.54,2.18,9.51,2.42
+L9.13,5.07 C8.52,5.32,7.96,5.66,7.44,6.05 L4.95,5.05
+C4.72,4.96,4.46,5.05,4.34,5.27 L2.34,8.73 C2.21,8.95,2.27,9.22,2.46,9.37
+L4.57,11.02 C4.53,11.34,4.5,11.67,4.5,12 S4.53,12.66,4.57,12.98 L2.46,14.63
+C2.27,14.78,2.22,15.05,2.34,15.27 L4.34,18.73 C4.46,18.95,4.73,19.03,4.95,18.95
+L7.44,17.95 C7.96,18.35,8.52,18.68,9.13,18.93 L9.51,21.58
+C9.54,21.82,9.75,22,10,22 L14,22 C14.25,22,14.46,21.82,14.49,21.58 L14.87,18.93
+C15.48,18.68,16.04,18.34,16.56,17.95 L19.05,18.95
+C19.28,19.04,19.54,18.95,19.66,18.73 L21.66,15.27
+C21.78,15.05,21.73,14.78,21.54,14.63 L19.43,12.98 Z M12,15.5
+C10.07,15.5,8.5,13.93,8.5,12 S10.07,8.5,12,8.5 S15.5,10.07,15.5,12
+S13.93,15.5,12,15.5 Z" />
+</vector>
\ No newline at end of file
diff --git a/res/layout/activity.xml b/res/layout/activity.xml
index 86c36e2..8e78f66 100644
--- a/res/layout/activity.xml
+++ b/res/layout/activity.xml
@@ -14,16 +14,33 @@
      limitations under the License.
 -->
 
-<androidx.viewpager.widget.ViewPager
+<LinearLayout
     xmlns:android="http://schemas.android.com/apk/res/android"
-    android:id="@+id/pager"
+    android:id="@+id/root"
     android:layout_width="match_parent"
-    android:layout_height="match_parent">
+    android:layout_height="match_parent"
+    android:orientation="vertical"
+    android:animateLayoutChanges="true">
 
-    <androidx.viewpager.widget.PagerTitleStrip
-        android:id="@+id/titles"
+    <Toolbar
+        android:id="@+id/toolbar"
         android:layout_width="match_parent"
-        android:layout_height="wrap_content"
-        android:layout_gravity="top" />
+        android:layout_height="?android:attr/actionBarSize"
+        android:background="@color/primary" />
 
-</androidx.viewpager.widget.ViewPager>
+    <androidx.viewpager.widget.ViewPager
+        android:id="@+id/pager"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent">
+
+        <androidx.viewpager.widget.PagerTitleStrip
+            android:id="@+id/titles"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_gravity="top"
+            android:background="@color/primary"
+            android:elevation="8dp" />
+
+    </androidx.viewpager.widget.ViewPager>
+
+</LinearLayout>
diff --git a/res/menu/activity.xml b/res/menu/activity.xml
index 74abfd3..28d044f 100644
--- a/res/menu/activity.xml
+++ b/res/menu/activity.xml
@@ -18,14 +18,16 @@
     <item
         android:id="@+id/menu_new_tab"
         android:title="@string/menu_new_tab"
-        android:icon="@android:drawable/ic_menu_add"
+        android:icon="@drawable/ic_menu_add"
         android:showAsAction="always" />
     <item
         android:id="@+id/menu_close_tab"
         android:title="@string/menu_close_tab"
-        android:icon="@android:drawable/ic_menu_close_clear_cancel"
-        android:showAsAction="ifRoom" />
+        android:icon="@drawable/ic_menu_close_clear_cancel"
+        android:showAsAction="always" />
     <item
         android:id="@+id/menu_item_settings"
-        android:title="@string/menu_item_settings" />
+        android:title="@string/menu_item_settings"
+        android:icon="@drawable/ic_menu_settings"
+        android:showAsAction="always" />
 </menu>
diff --git a/res/values/arrays.xml b/res/values/arrays.xml
index 1c83e9b..5349af6 100644
--- a/res/values/arrays.xml
+++ b/res/values/arrays.xml
@@ -49,13 +49,15 @@
         <item>Green text on black</item>
         <item>Amber text on black</item>
         <item>Red text on black</item>
+        <item>Blue text on black</item>
     </string-array>
     <string-array name="text_colors_values" translatable="false">
-        <item>#000000/#ffffff</item>
-        <item>#f0f0f0/#000000</item>
-        <item>#00f000/#000000</item>
-        <item>#f0c000/#000000</item>
-        <item>#f00000/#000000</item>
+        <item>#000000/#FFFFFF</item>
+        <item>#FFFFFF/#000000</item>
+        <item>#4CAF50/#000000</item>
+        <item>#FF9800/#000000</item>
+        <item>#F44336/#000000</item>
+        <item>#3F51B5/#000000</item>
     </string-array>
 
 </resources>
diff --git a/res/values/colors.xml b/res/values/colors.xml
new file mode 100644
index 0000000..555f1c5
--- /dev/null
+++ b/res/values/colors.xml
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright (C) 2015 The CyanogenMod Project
+     Copyright (C) 2015 The LineageOS Project
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+-->
+<resources>
+    <!-- Material -->
+    <color name="primary">#212121</color>
+    <color name="primary_dark">#0c0c0c</color>
+    <color name="accent">#009688</color>
+
+    <color name="white">#ffffff</color>
+</resources>
diff --git a/res/values/styles.xml b/res/values/styles.xml
new file mode 100644
index 0000000..7c8cb8f
--- /dev/null
+++ b/res/values/styles.xml
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright (C) 2015 The CyanogenMod Project
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+-->
+<resources>
+    <style name="TermTheme" parent="@android:Theme.Material">
+        <item name="android:colorPrimary">@color/primary</item>
+        <item name="android:colorPrimaryDark">@color/primary_dark</item>
+        <item name="android:colorAccent">@color/accent</item>
+    </style>
+    <style name="TermTheme.NoActionBar">
+        <item name="android:windowNoTitle">true</item>
+        <item name="android:windowActionBar">false</item>
+    </style>
+</resources>
diff --git a/res/xml/settings.xml b/res/xml/settings.xml
index ea85402..5a7a361 100644
--- a/res/xml/settings.xml
+++ b/res/xml/settings.xml
@@ -41,7 +41,8 @@
             android:summary="%s"
             android:dialogTitle="@string/font_size_title"
             android:entries="@array/font_size_labels"
-            android:entryValues="@array/font_size_values" />
+            android:entryValues="@array/font_size_values"
+            android:defaultValue="12" />
         <ListPreference
             android:key="text_colors"
             android:title="@string/text_colors_title"
diff --git a/src/com/android/terminal/TerminalActivity.java b/src/com/android/terminal/TerminalActivity.java
index e23b862..ecef6fd 100644
--- a/src/com/android/terminal/TerminalActivity.java
+++ b/src/com/android/terminal/TerminalActivity.java
@@ -19,6 +19,7 @@ package com.android.terminal;
 import static com.android.terminal.Terminal.TAG;
 
 import android.Manifest;
+import android.animation.LayoutTransition;
 import android.app.Activity;
 import android.content.ComponentName;
 import android.content.Context;
@@ -40,6 +41,7 @@ import android.view.Menu;
 import android.view.MenuItem;
 import android.view.View;
 import android.view.ViewGroup;
+import android.widget.Toolbar;
 
 /**
  * Activity that displays all {@link Terminal} instances running in a bound
@@ -195,6 +197,9 @@ public class TerminalActivity extends Activity {
 
         setContentView(R.layout.activity);
 
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setActionBar(toolbar);
+
         mPager = (ViewPager) findViewById(R.id.pager);
         mTitles = (PagerTitleStrip) findViewById(R.id.titles);
 
@@ -203,6 +208,9 @@ public class TerminalActivity extends Activity {
         View decorView = getWindow().getDecorView();
         decorView.setOnSystemUiVisibilityChangeListener(mUiVisibilityChangeListener);
 
+        ViewGroup root = (ViewGroup) findViewById(R.id.root);
+        root.getLayoutTransition().enableTransitionType(LayoutTransition.CHANGING);
+
         final int REQUEST_WRITE_STORAGE = 51;
 
         if (checkSelfPermission(android.Manifest.permission.WRITE_EXTERNAL_STORAGE)
diff --git a/src/com/android/terminal/TerminalSettingsActivity.java b/src/com/android/terminal/TerminalSettingsActivity.java
index 013c0f2..f7d49d9 100644
--- a/src/com/android/terminal/TerminalSettingsActivity.java
+++ b/src/com/android/terminal/TerminalSettingsActivity.java
@@ -23,7 +23,6 @@ import android.preference.PreferenceActivity;
 import android.preference.PreferenceScreen;
 import android.preference.SwitchPreference;
 import android.util.Log;
-import android.view.Menu;
 import android.view.MenuItem;
 
 import static com.android.terminal.Terminal.TAG;
@@ -52,10 +51,17 @@ public class TerminalSettingsActivity extends PreferenceActivity {
         mScreenOrientationPref = (ListPreference) findPreference(KEY_SCREEN_ORIENTATION);
         mFontSizePref = (ListPreference) findPreference(KEY_FONT_SIZE);
         mTextColorsPref = (ListPreference) findPreference(KEY_TEXT_COLORS);
+
+        getActionBar().setDisplayHomeAsUpEnabled(true);
     }
 
     @Override
     public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                finish();
+                return true;
+        }
         return super.onOptionsItemSelected(item);
     }
 }
diff --git a/src/com/android/terminal/TerminalView.java b/src/com/android/terminal/TerminalView.java
index 6500d75..c5a8b63 100644
--- a/src/com/android/terminal/TerminalView.java
+++ b/src/com/android/terminal/TerminalView.java
@@ -373,9 +373,9 @@ public class TerminalView extends ListView {
         val = sp.getString(TerminalSettingsActivity.KEY_FONT_SIZE, "12");
         mMetrics.setTextSize(ptToDp(Float.parseFloat(val)));
 
-        val = sp.getString(TerminalSettingsActivity.KEY_TEXT_COLORS, "black/white");
-        int fg = 0x000000;
-        int bg = 0xffffff;
+        val = sp.getString(TerminalSettingsActivity.KEY_TEXT_COLORS, "white/black");
+        int fg = 0xfafafa;
+        int bg = 0x212121;
         int idx = val.indexOf('/');
         if (idx != -1) {
             try {
-- 
2.17.1

